contracts/src/fp/OPSuccinctFaultDisputeGame.sol

```diff
@@ -377,13 +438,19 @@ contract OPSuccinctFaultDisputeGame is Clone, ISemver {
         if (parentGameStatus == GameStatus.IN_PROGRESS) revert ParentGameNotResolved();
 
         // INVARIANT: If the parent game's claim is invalid, then the current game's claim is invalid.
+        // INVARIANT: If there is no challenger, i.e. `claimData.counteredBy == address(0)`, the bond is burnt.
         if (parentGameStatus == GameStatus.CHALLENGER_WINS) {
             claimData.status = ProposalStatus.Resolved;
             status = GameStatus.CHALLENGER_WINS;
             resolvedAt = Timestamp.wrap(uint64(block.timestamp));
 
-            // Record the challenger's reward
-            credit[claimData.counteredBy] += address(this).balance;
+            // Add challenger's reward to the entry point contract.
+            (bool success,) = ENTRY_POINT.call{value: address(this).balance}(
+                abi.encodeWithSelector(
+                    OPSuccinctEntryPoint.addCredit.selector, claimData.counteredBy, address(this).balance
+                )
+            );
+            if (!success) revert BondTransferFailed();
 
             emit Resolved(status);
 
```

Comment:
I'd add a comment after this function that at this point we know `parentGameStatus = GameStatus.DEFENDER_WINS`.